src/deadline/client/cli/_groups/job_group.py

```diff
@@ -706,6 +724,307 @@ def job_download_output(step_id, task_id, output, **args):
+                timeout_info = f" [{elapsed_time:.1f}s elapsed, {remaining:.1f}s remaining]"
+            else:
+                timeout_info = f" [{elapsed_time:.1f}s elapsed]"
+
+            # Clear the current line and update in place
+            click.echo(
+                f"\rCurrent status: {status}.{timeout_info}",
+                nl=False,
+            )
+
+    if not is_json_output:
+        click.echo(f"Waiting for job {job_id} to complete...")
+
+    try:
+        result = api.wait_for_job_completion(
+            farm_id=farm_id,
+            queue_id=queue_id,
+            job_id=job_id,
+            max_poll_interval=max_poll_interval,
+            timeout=timeout,
+            config=config,
+            status_callback=status_callback,
+        )
+
+        if is_json_output:
+            # Return everything as JSON
+            response = {
+                "status": result.status,
+                "elapsedTime": result.elapsed_time,
+                "failedTasks": [
+                    {
+                        "stepId": task.step_id,
+                        "taskId": task.task_id,
+                        "stepName": task.step_name,
+                        "sessionId": task.session_id,
+                    }
+                    for task in result.failed_tasks
+                ],
+            }
+            click.echo(json.dumps(response, indent=2))
+        else:
+            # Use verbose output with YAML formatting
+            click.echo(f"Job completed with status: {result.status}")
+            click.echo(f"Elapsed time: {result.elapsed_time:.1f} seconds")
+
+            if result.failed_tasks:
+                click.echo(f"Found {len(result.failed_tasks)} failed tasks:")
+                failed_tasks_dict = [
+                    {
+                        "stepId": task.step_id,
+                        "taskId": task.task_id,
+                        "stepName": task.step_name,
+                        "sessionId": task.session_id,
+                    }
+                    for task in result.failed_tasks
+                ]
+                click.echo(_cli_object_repr(failed_tasks_dict))
+            else:
+                click.echo("No failed tasks found.")
+
+        # Determine exit code based on job status
+        exit_code = 0
+        if result.status == "SUCCEEDED" and not result.failed_tasks:
+            exit_code = 0
+        elif result.status == "FAILED" or result.failed_tasks:
+            exit_code = 2
+        elif result.status == "CANCELED":
+            exit_code = 3
+        elif result.status == "ARCHIVED":
+            exit_code = 4
+        elif result.status == "NOT_COMPATIBLE":
+            exit_code = 5
+        else:
+            # Unknown status, treat as failure
+            exit_code = 2
+
+        sys.exit(exit_code)
+
+    except DeadlineOperationError as e:
+        # Check if this is a timeout error
+        if "Timeout waiting for job" in str(e):
```

Comment:
In https://github.com/aws-deadline/deadline-cloud/blob/mainline/src/deadline/client/exceptions.py we have `DeadlineOperationError` and `DeadlineOperationCanceled`. Maybe there could be another `DeadlineOperationTimedOut` to model this case explicitly instead of depending on the contents of the message?